test/cqlpy/test_custom_index.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2025-present ScyllaDB
+#
+# SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+
+###############################################################################
+# Tests for custom indexes
+###############################################################################
+
+import pytest
+from .util import new_test_table
+from cassandra.protocol import InvalidRequest
+
+
+def test_create_vector_search_index(cql, test_keyspace):
+    schema = 'p int primary key, v vector<float, 3>'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        cql.execute(f"CREATE CUSTOM INDEX ON {table}(v) USING 'opensearch_vector_index'")
+
+def test_create_vector_search_index_with_usearch_engine(cql, test_keyspace):
+    schema = 'p int primary key, v vector<float, 3>'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        cql.execute(f"CREATE CUSTOM INDEX ON {table}(v) USING 'usearch_vector_index'")
+
+def test_create_vector_search_index_without_custom_keyword(cql, test_keyspace):
+    schema = 'p int primary key, v vector<float, 3>'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        cql.execute(f"CREATE INDEX ON {table}(v) USING 'opensearch_vector_index'")
+
+def test_create_custom_index_with_invalid_class(cql, test_keyspace):
+    schema = 'p int primary key, v vector<float, 3>'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        invalid_custom_class = "invalid.custom.class"
+        with pytest.raises(InvalidRequest):
+            cql.execute(f"CREATE CUSTOM INDEX invalid_idx ON {table}(v) USING '{invalid_custom_class}'")
+        with pytest.raises(InvalidRequest):
+            cql.execute(f"CREATE CUSTOM INDEX no_class_idx ON {table}(v)")
+
+def test_create_vector_search_index_on_nonvector_column(cql, test_keyspace):
+    schema = 'p int primary key, v int'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        with pytest.raises(InvalidRequest):
+            cql.execute(f"CREATE CUSTOM INDEX invalid_idx ON {table}(v) USING 'opensearch_vector_index'")
+        with pytest.raises(InvalidRequest):
+            cql.execute(f"CREATE CUSTOM INDEX invalid_idx ON {table}(v) USING 'usearch_vector_index'")
+        
+
+def is_scylla(cql):
+    return any('scylla' in name for name in [row.table_name for row in cql.execute("SELECT * FROM system_schema.tables WHERE keyspace_name = 'system'")])
```

Comment:
You can import is_scylla from .util instead of having another copy here. @denesb added this function to util.py in 6e53264bc3cf1abcc9621a3e5296c07471c14cd8